index.js

```diff
@@ -1,2 +1,2 @@
 'use strict';
-module.exports = /^#!(.*)/;
+module.exports = /^#![ \t]*([^\s]+)[ \t]*([^\s]*)?.*$/m;
```

Comment:
Why `m`? It shouldn't try to match stuff on any line, just the first.

Reply:
I also think it should be made a little bit stricter. The binary can only be `\w`, so `([^\s]*)` => `(\w*)`

Reply:
Can you also create a group for the executable arguments?

Reply:
> Why m? It shouldn't try to match stuff on any line, just the first.

You are correct, multiline flag is not needed. Mostly forgot that `.*` does not match newline.

> I also think it should be made a little bit stricter. The binary can only be \w, so ([^\s]*) => (\w*)

The possible characters in a binary name is not quite that narrow. At very least a dash is possible. Probably a period too.

> Can you also create a group for the executable arguments?

Sure, it might make group two and three converge a bit. Any ideas on how to avoid that?

Reply:
> The possible characters in a binary name is not quite that narrow. At very least a dash is possible. Probably a period too.

Ok, include that too.

> Sure, it might make group two and three converge a bit. Any ideas on how to avoid that?

Not sure what you mean. The binary name would be a separate group from the executable arguments and would not be included in the binary name group.

```
#!/usr/bin/env node --foo
      ^          ^      ^
    group 1  group 2  group 3
```